scarb/tests/build.rs

```diff
@@ -687,3 +687,58 @@ fn edition_must_exist() {
                  unknown variant `2021`, expected one of `2023_01`, `2023_10`, `2023_11`
         "#});
 }
+
+#[test]
+fn dev_dep_used_outside_tests() {
+    let t = TempDir::new().unwrap();
+    let q = t.child("q");
+    ProjectBuilder::start()
+        .name("q")
+        .lib_cairo(
+            r#"
+            fn dev_dep_function() -> felt252 {
+                42
+            }
+            "#,
+        )
```

Comment:
Nit: can fit to a single line
```suggestion
        .lib_cairo("fn dev_dep_function() -> felt252 { 42 }")
```